### PR TITLE
Wip/m.otto@phytec.de/board config 1.1

### PR DESCRIPTION
**Pull request by motto-phytec**
Update phytec-board-config tool

- add version for the tool
- rework awsclient restart after onboarding
- if onboarding not finished, then deactivate awsclient service
- add restart awsclient as parameter
- add shadow commands, which necessary for awsclient from 1.4

### CODE REVIEW

File: recipes-support/provisioning/files/phytec-board-config.sh [hunk 1]
```diff
@@ -338,7 +339,6 @@ The next steps are:
  4) Check the state of your device in
     your IoT Device Suite account" $WT_HEIGHT $WT_WIDTH
 		fi
-		do_restart_awsclient
```
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: Could you explain the reasoning behind always restarting the awsclient when onboarding?

Why is the additional option to restart the awsclient needed? Can't you just use the `systemctl` command?

Can the notice about manually restarting the awsclient be removed after this change?

File: recipes-support/provisioning/files/phytec-board-config.sh [hunk 2]
```diff
@@ -57,8 +60,8 @@ calc_wt_size() {
 
 do_about() {
 	whiptail --msgbox "\
-This tool provides a straightforward way of doing initial
-configuration of your PHYTEC board for the
+This tool (version ${version})  provides a straightforward
```
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: I do not see the need to add the version number here. If it is needed one can just call the program with the dedicated option you added. Maybe add the version to the `--backtitle` of the main setup window, alongside the device tree model name.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
Please rephrase you commit message. It reads like you added a version to the awsclient, which does not seem to be intended.

File: recipes-support/provisioning/files/phytec-board-config.sh (hunk 1 above)
Comment (motto-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: The three times restart of the awsclient is necessary to download all the configuration files and to finalize the onboarding process. 
The additional option to restart the awsclient is needed for the following case:
if the user generates a token or generates a new account, but do not add the token directly on the platform or do not activate the account directly, then the onboarding process is not finished. In the old version the board shows onboarding successfully, what is not correct. In the new version the board shows not onboarded. With the new option   you can finalize the onboarding process. The awsclient will be restarted and in the config is written, that the onboarding process is finished. The config information is used to show the right onboarding state at board booting.

File: recipes-support/provisioning/files/phytec-board-config.sh (hunk 2 above)
Comment (motto-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: i remove the version at this point

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**